util/Logger.h

```diff
@@ -1,56 +1,292 @@
+            with the code to understand.
+            For example reporting that the network disconnected due to a client initiated shutdown
+            with a linger time of 30 ms before closing.
+    trace - used for the most detailed logging.  Trace should probably only be used with a named
+            logger, since the extreme detail will only be of interest to a developer working on a
+            specific section of code and without the additional filtering of the name the log
+            output will be flooded.  Setting all logger's thresholds to trace may generate
+            unreasonably large log files.
+            For example the network logger might log every message received/sent as it is
+            added to or removed from its internal queues.
+
+    Usage:
+
+    Generating Logs:
+
+    The default or unnamed logger works as an output stream, like std::cout as follows:
+
+    ErrorLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the error threshold or greater.";
+    WarnLogger()  << "Put any streamable output here.  It will only be
+                  << "computed at the warn threshold or greater.";
+    InfoLogger()  << "Put any streamable output here.  It will only be
+                  << "computed at the info threshold or greater.";
+    DebugLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the debug threshold or greater.";
+    TraceLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the trace threshold or greater.";
+
+    The named loggers are created with:
+
+    CreateThreadedLogger(name_of_logger);
+
+    And used with
+
+    ErrorLogger(name_of_logger) << "any streamable output, only computed "
+                                << "at the error threshold or greater "
+                                << "for the 'name_of_logger' logger.";
+    WarnLogger(name_of_logger)  << "streamable output";
+    InfoLogger(name_of_logger)  << "streamable output";
+    DebugLogger(name_of_logger) << "streamable output";
+    TraceLogger(name_of_logger) << "streamable output";
+
+    <Error/Warn/Info/Debug/Trace>Logger() is a macro that gates the
+    generation of the ""streamable output" following the stream
+    operator. If the "streamable output" is a function call it is only
+    executed if that given source is above its threshold and it has a sink
+    to consume the generated log.
+
+
+    Setting Logger Thresholds:
+
+    Set the logger threshold with:
+
+    SetLoggerThreshold(name_of_logger, a_threshold);
+
+    a_threshold is written LogLevel::error, etc. Use "" for the name of the unnamed source.
+
+
+    Force the threshold for all loggers to be the same with:
+
+    OverrideAllLoggerThresholds(a_threshold);
+
+    These functions are used to provide further options for setting logger thresholds using
```

Comment:
So most coders shouldn't use them then, and they're really implementation details?